python_modules/libraries/dagster-airflow/dagster_airflow/dagster_factory.py

```diff
@@ -0,0 +1,190 @@
+    contains_duplicate_task_names,
+    create_airflow_connections,
+)
+
+
+def make_dagster_definitions_from_airflow_dag_bag(
+    dag_bag: DagBag,
+    connections: Optional[List[Connection]] = None,
+):
+    """Construct a Dagster definition corresponding to Airflow DAGs in DagBag.
+
+    Usage:
+        Create `make_dagster_definition.py`:
+            from dagster_airflow import make_dagster_definition_from_airflow_dag_bag
+            from airflow_home import my_dag_bag
+
+            def make_definition_from_dag_bag():
+                return make_dagster_definition_from_airflow_dag_bag(my_dag_bag)
+
+        Use Definitions as usual, for example:
+            `dagit -f path/to/make_dagster_definition.py`
+
+    Args:
+        dag_bag (DagBag): Airflow DagBag Model
+        connections (List[Connection]): List of Airflow Connections to be created in the Airflow DB
+
+    Returns:
+        Definitions
+    """
+    schedules, jobs = make_schedules_and_jobs_from_airflow_dag_bag(
+        dag_bag,
+        connections,
+    )
+
+    return Definitions(
+        schedules=schedules,
+        jobs=jobs,
+    )
+
+
+def make_dagster_definitions_from_airflow_dags_path(
+    dag_path: str,
+    safe_mode: bool = True,
+    connections: Optional[List[Connection]] = None,
+):
+    """Construct a Dagster repository corresponding to Airflow DAGs in dag_path.
+
+    Usage:
+        Create ``make_dagster_definitions.py``:
+
+        .. code-block:: python
+
+            from dagster_airflow import make_dagster_definitions_from_airflow_dags_path
+
+            def make_definitions_from_dir():
+                return make_dagster_definitions_from_airflow_dags_path(
+                    '/path/to/dags/',
+                )
+
+        Use RepositoryDefinition as usual, for example:
+        ``dagit -f path/to/make_dagster_repo.py -n make_repo_from_dir``
+
+    Args:
+        dag_path (str): Path to directory or file that contains Airflow Dags
+        include_examples (bool): True to include Airflow's example DAGs. (default: False)
+        safe_mode (bool): True to use Airflow's default heuristic to find files that contain DAGs
+            (ie find files that contain both b'DAG' and b'airflow') (default: True)
+        connections (List[Connection]): List of Airflow Connections to be created in the Airflow DB
+
+    Returns:
+        Definitions
+    """
+    check.str_param(dag_path, "dag_path")
+    check.bool_param(safe_mode, "safe_mode")
+    connections = check.opt_list_param(connections, "connections", of_type=Connection)
+    # add connections to airflow so that dag evaluation works
+    create_airflow_connections(connections)
+    try:
+        dag_bag = DagBag(
+            dag_folder=dag_path,
+            include_examples=False,  # Exclude Airflow example dags
+            safe_mode=safe_mode,
+        )
+    except Exception:
+        raise DagsterAirflowError("Error initializing airflow.models.dagbag object with arguments")
```

Comment:
Don't we need to set an inner exception or something to not eat invaluable error information? I cannot recall the current state of such things in all version of python we suppot.